### PR TITLE
Expand the documentation of map_to_primitive

### PR DESCRIPTION
**Pull request by plexus**
Explain in more detail why you would want to use iso8601, and how.

### CODE REVIEW

File: yaks/README.md [hunk 1]
```diff
@@ -839,17 +839,38 @@ end
...
+
+```ruby
+Yaks.new do
+  map_to_primitive Date, Time, DateTime, ActiveSupport::TimeWithZone, &:iso8601
+end
+```
+
+`map_to_primitive` can also be used to transform alternative data
+structures, like those from [Hamster](https://github.com/hamstergem/hamster),
+into Ruby arrays and hashes. Use `call()` to recursively turn things into
```
Comment (groyoh) [inline yaks/README.md]: Here you mean "use `call` inside the block", right? Wouldn't it be better to be a bit more explicit and give an example?

**Comment by groyoh** [comment]
The rubocop version was updated and the rubocop task is now failling.

File: yaks/README.md (hunk 1 above)
Comment (plexus) [inline yaks/README.md]: :+1: sure I can do that :)

Comment (plexus) [inline yaks/README.md]: Actually there *is* an example right after 

```ruby
Yaks.new do
  map_to_primitive Hamster::Vector, Hamster::List do |list|
    list.map do |item|
      call(item)
    end
  end
end
```

Comment (groyoh) [inline yaks/README.md]: My bad then :smile:

**Comment by groyoh** [comment]
Nice job! I'll merge it as soon as you fix the rubocop errors :wink:

**Comment by plexus** [comment]
Was already on it

**Pull request merged**